LeaderboardBackend/Services/Impl/UserService.cs

```diff
@@ -98,4 +99,23 @@ public async Task<CreateUserResult> CreateUser(RegisterRequest request)
 
         return newUser;
     }
+
+    public async Task<UpdateUserResult> UpdateUser(Guid id, UpdateUserRequest request)
+    {
+        User? user = await applicationContext.Users.SingleOrDefaultAsync(user => user.Id == id);
+
+        if (user is null)
+        {
+            return new UserNotFound();
+        }
+
+        if (user.Role == UserRole.Administrator)
+        {
+            return new BadRole();
```

Comment:
It doesn't make sense to return BadRole here because there is no role that's allowed to perform this operation.

Reply:
If we don't have this though, an admin can ban another.

Reply:
I just mean "BadRole" doesn't really make sense as a response here. In fact, you could argue that this is more akin to something that would produce a 422 error, since it's basically an invalid value.

Reply:
But it's not an invalid enum value, though. We want to block admins from being banned. I'd say that's a 403